### PR TITLE
wxPython 4 migration (WIP)

### PR DESCRIPTION
**Pull request by 0x00b1**
macOS instructions:

- Uninstall Homebrew `wxmac` and `wxpython`:

```
$ brew uninstall wxmac wxpython
```

- Grab the latest wxPython from PyPI:

```
$ pip2 install -e .
```

- Run CellProfiler with:

```sh
$ python2 cellprofiler/__main__.py
```

### CODE REVIEW

**Comment by 0x00b1** [comment]
Useful links

https://wxpython.org/Phoenix/docs/html/classic_vs_phoenix.html

https://github.com/robmcmullen/omnivore/wiki/wxPython-Phoenix-conversion

File: .travis.yml [hunk 1]
```diff
@@ -1,9 +1,5 @@
-addons:
-  apt:
-    packages:
-      - libhdf5-serial-dev
-      - python-mysqldb
-      - python-pip
+before_install:
+  - sudo apt-get install --yes libgstreamer-plugins-base1.0-0 libgstreamer-plugins-base1.0-dev libgstreamer1.0-0 libgstreamer1.0-dev libgtk-3-0 libgtk-3-dev libhdf5-serial-dev libjpeg-dev libnotify-dev libpng-dev libegl1-mesa-dev libgles2-mesa-dev libsdl2-dev libsm-dev libsm6 libtiff5 libtiff5-dev libwebkitgtk-3.0-dev freeglut3 freeglut3-dev python-dev python-mysqldb python-pip
```
Comment (AetherUnbound) [inline .travis.yml]: Is there a reason this is specified inline rather than as a list?

File: cellprofiler/gui/app.py [hunk 2]
```diff
@@ -57,6 +56,8 @@ def __init__(self, *args, **kwargs):
     def OnInit(self):
         import cellprofiler.gui.cpframe
 
+        wx.lib.inspection.InspectionTool().Show()
```
Comment (AetherUnbound) [inline cellprofiler/gui/app.py]: Is this still needed? It looks like a debug thing (see also the import above).

File: cellprofiler/gui/artist.py [hunk 3]
```diff
@@ -326,22 +329,22 @@ def overlay(self):
         if self.__overlay is not None:
             return self.__overlay
         sm = matplotlib.cm.ScalarMappable(cmap=self.colormap)
-        sm.set_clim(vmin=1, vmax=numpy.max([numpy.max(l) for l in self.labels]) + 1)
+        sm.set_clim(vmin=1, vmax=numpy.max([numpy.max(label) for label in self.labels]) + 1)
```
Comment (AetherUnbound) [inline cellprofiler/gui/artist.py]: What was this even doing? haha

File: cellprofiler/gui/cpframe.py [hunk 4]
```diff
@@ -150,10 +153,12 @@ def __init__(self, *args, **kwds):
 
         """
         kwds["style"] = wx.DEFAULT_FRAME_STYLE
-        wx.Frame.__init__(self, *args, **kwds)
+
         self.__pipeline = cellprofiler.gui.pipeline.Pipeline()
-        self.__workspace = cellprofiler.gui.workspace.Workspace(
-                self.__pipeline, None, None, None, None, None)
+        self.__workspace = cellprofiler.gui.workspace.Workspace(self.__pipeline, None, None, None, None, None)
```
Comment (AetherUnbound) [inline cellprofiler/gui/cpframe.py]: Not enough `None`

File: cellprofiler/gui/cpframe.py [hunk 5]
```diff
@@ -534,68 +523,68 @@ def __add_menu(self):
...
+            helpString="Reload the project file, discarding changes")
         submenu = wx.Menu()
         submenu.Append(
-                ID_FILE_LOAD_PIPELINE,
-                'Pipeline from File...',
-                'Import a pipeline into the project from a .%s file' %
-                cellprofiler.preferences.EXT_PIPELINE)
+            ID_FILE_LOAD_PIPELINE,
+            'Pipeline from File...',
+            'Import a pipeline into the project from a .%s file' %
```
Comment (AetherUnbound) [inline cellprofiler/gui/cpframe.py]: Do you mind changing this to `.format` as well?

File: requirements.txt [hunk 6]
```diff
@@ -17,3 +17,4 @@ requests
 scikit-image
 scikit-learn
 scipy
+wxPython==4.0.3
```
Comment (AetherUnbound) [inline requirements.txt]: This might be wxPython>=4.0 :thinking: 
Just to make it more open!

File: setup.py [hunk 7]
```diff
@@ -77,7 +77,8 @@ def find_resources(directory, subdirectory):
         "requests",
         "scikit-image",
         "scikit-learn",
-        "scipy"
+        "scipy",
+        "wxPython==4.0.3"
```
Comment (AetherUnbound) [inline setup.py]: Same here, maybe wxPython>=4.0!

**Comment by AetherUnbound** [comment]
@0x00b1 is this PR ready?

**Comment by 0x00b1** [comment]
@AetherUnbound No! 😂

@0x00b1 rebase ASAP!

**Pull request merged**